### DISCUSSION TITLE
Why Redis?

### DISCUSSION THREAD

**Discussion by MOHITRAJANI** [bee-queue/bee-queue#652] [Q&A]
Hey Community, I have a very generic question for the planners and all contributors of such a great library. That question is, why is Bee-Queue backed with REDIS In-Memory Database?
It could have been any other database also right? Probably MySQL or maybe even MongoDb?
I ask this question because I am currently implementing my Thesis and this question was raised to me when I had to choose a Database solution for my architecture.

If someone could list down a few points that lead to this decision I would be obliged to them.

Thanks in advance.

**Comment by compwright** [comment]
I am not the creator of this library, but the choice of Redis makes sense.

* MongoDB doesn't support atomic operations. Redis does. This is important for dealing with concurrency.
* MySQL doesn't provide good support for structured JSON data before version 8.
* Generally, hosting Redis tends to be less resource intensive and therefore cheaper than MySQL (and probably MongoDB).
* Neither MongoDB nor MySQL support pub/sub which is used to report results back on jobs run.
* In-memory means it runs faster than a transactional database would typically run, reducing the bottleneck impact of bee-queue on an application overall.

See https://web.archive.org/web/20220504052812/https://github.com/bee-queue/bee-queue/wiki/Origin for what the original authors were thinking when they wrote this library.
